### DISCUSSION TITLE
Show my name instead of username anywhere

### DISCUSSION THREAD

**Discussion by exportio** [community/community#57162] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Why is it asking me to set my name if it's only going to use my username everywhere?  Like when I report an issue or respond to a comment.

Even on this post you see the username instead of my name.

**Comment by danielalyoshin** [comment, marked as answer]
Assuming you mean your profile display name, that is simply for show. You can choose an alias, use your real name, or not set a display name at all.

Hope this helps!

**Reply by mr-jolly** [reply]
This is incorrect. Your activity on GitHub **does not** use the display name - it always displays username, which is incredibly inconvenient and can not be worked around if you have connected your GitHub account to an organization.

**Reply by danielalyoshin** [reply]
I never claimed that your activity on GitHub uses your display name. I said that the display name is just for show.

Hope this clears things up.

**Reply by mr-jolly** [reply]
Then your response is a non-answer. You have simply described the broken GitHub functionality.

**Comment by rishabhkemni** [comment]
https://docs.github.com/articles/changing-your-github-username#:~:text=In%20the%20upper%2Dright%20corner,%22%20section%2C%20click%20Change%20username.

https://superuser.com/questions/834247/how-can-i-make-github-for-windows-use-my-real-name-for-commits

https://github.com/orgs/community/discussions/10840

I hope above 3 links are self explanatory

**Reply by mr-jolly** [reply]
> https://docs.github.com/articles/changing-your-github-username#:~:text=In%20the%20upper%2Dright%20corner,%22%20section%2C%20click%20Change%20username.

_Changing your username can have unintended side effects._ 

I don't want to change my username. I want GitHub to respect and use the display name I have set.

> https://superuser.com/questions/834247/how-can-i-make-github-for-windows-use-my-real-name-for-commits

This link centers on the name set by git config - as long as you are using native git functionality, this name is used. It is the non-native functionality - added by GitHub - pull requests for example, that neither honors the name specified in git config nor the GitHub display name.
> https://github.com/orgs/community/discussions/10840

![image](https://github.com/community/community/assets/2116971/f515424a-f272-4011-9bf7-67beb593e77f)

This is another unanswered by GitHub discussion of how their implementation sucks and how display name should be displayed, not username.

> I hope above 3 links are self explanatory

Self explanatory, yes. Helpful? Not at all.

**Comment by HaydenReeve** [comment]
Probably worth subscribing and upvoting this discussion here.

https://github.com/orgs/community/discussions/10840

Github currently doesn't support this which I feel is a real issue.
